{"id":"../../utils/catchAsync.js","dependencies":[{"name":"C:\\Users\\TEYI\\Desktop\\Formation NodeJs\\Formation Jonas\\natours\\package.json","includedInParent":true,"mtime":1590712966311}],"generated":{"js":"/**\r\n * Elle nous permet de se passer du try and catch block ds nos fnction et de la maniere autonome de gerer les erreurs\r\n */\nmodule.exports = function (fn) {\n  return function (req, res, next) {\n    fn(req, res, next).catch(next);\n  };\n};"},"sourceMaps":{"js":{"mappings":[{"generated":{"line":1,"column":0},"source":"../../utils/catchAsync.js","original":{"line":1,"column":0}},{"name":"module","generated":{"line":4,"column":0},"source":"../../utils/catchAsync.js","original":{"line":5,"column":0}},{"generated":{"line":4,"column":6},"source":"../../utils/catchAsync.js","original":{"line":5,"column":6}},{"name":"exports","generated":{"line":4,"column":7},"source":"../../utils/catchAsync.js","original":{"line":5,"column":7}},{"generated":{"line":4,"column":14},"source":"../../utils/catchAsync.js","original":{"line":5,"column":0}},{"generated":{"line":4,"column":17},"source":"../../utils/catchAsync.js","original":{"line":5,"column":17}},{"name":"fn","generated":{"line":4,"column":27},"source":"../../utils/catchAsync.js","original":{"line":5,"column":17}},{"generated":{"line":4,"column":29},"source":"../../utils/catchAsync.js","original":{"line":5,"column":19}},{"generated":{"line":4,"column":31},"source":"../../utils/catchAsync.js","original":{"line":5,"column":23}},{"generated":{"line":5,"column":0},"source":"../../utils/catchAsync.js","original":{"line":7,"column":4}},{"generated":{"line":5,"column":9},"source":"../../utils/catchAsync.js","original":{"line":7,"column":11}},{"name":"req","generated":{"line":5,"column":19},"source":"../../utils/catchAsync.js","original":{"line":7,"column":12}},{"generated":{"line":5,"column":22},"source":"../../utils/catchAsync.js","original":{"line":7,"column":11}},{"name":"res","generated":{"line":5,"column":24},"source":"../../utils/catchAsync.js","original":{"line":7,"column":17}},{"generated":{"line":5,"column":27},"source":"../../utils/catchAsync.js","original":{"line":7,"column":11}},{"name":"next","generated":{"line":5,"column":29},"source":"../../utils/catchAsync.js","original":{"line":7,"column":22}},{"generated":{"line":5,"column":33},"source":"../../utils/catchAsync.js","original":{"line":7,"column":11}},{"generated":{"line":5,"column":35},"source":"../../utils/catchAsync.js","original":{"line":7,"column":31}},{"name":"fn","generated":{"line":6,"column":0},"source":"../../utils/catchAsync.js","original":{"line":8,"column":8}},{"name":"fn","generated":{"line":6,"column":4},"source":"../../utils/catchAsync.js","original":{"line":8,"column":8}},{"generated":{"line":6,"column":6},"source":"../../utils/catchAsync.js","original":{"line":8,"column":10}},{"name":"req","generated":{"line":6,"column":7},"source":"../../utils/catchAsync.js","original":{"line":8,"column":11}},{"generated":{"line":6,"column":10},"source":"../../utils/catchAsync.js","original":{"line":8,"column":10}},{"name":"res","generated":{"line":6,"column":12},"source":"../../utils/catchAsync.js","original":{"line":8,"column":17}},{"generated":{"line":6,"column":15},"source":"../../utils/catchAsync.js","original":{"line":8,"column":10}},{"name":"next","generated":{"line":6,"column":17},"source":"../../utils/catchAsync.js","original":{"line":8,"column":22}},{"generated":{"line":6,"column":21},"source":"../../utils/catchAsync.js","original":{"line":8,"column":10}},{"generated":{"line":6,"column":22},"source":"../../utils/catchAsync.js","original":{"line":8,"column":8}},{"name":"catch","generated":{"line":6,"column":23},"source":"../../utils/catchAsync.js","original":{"line":8,"column":28}},{"generated":{"line":6,"column":28},"source":"../../utils/catchAsync.js","original":{"line":8,"column":8}},{"name":"next","generated":{"line":6,"column":29},"source":"../../utils/catchAsync.js","original":{"line":8,"column":34}},{"generated":{"line":6,"column":33},"source":"../../utils/catchAsync.js","original":{"line":8,"column":8}},{"generated":{"line":7,"column":0},"source":"../../utils/catchAsync.js","original":{"line":9,"column":5}},{"generated":{"line":7,"column":3},"source":"../../utils/catchAsync.js","original":{"line":7,"column":4}},{"generated":{"line":8,"column":0},"source":"../../utils/catchAsync.js","original":{"line":10,"column":1}},{"generated":{"line":8,"column":1},"source":"../../utils/catchAsync.js","original":{"line":5,"column":0}}],"sources":{"../../utils/catchAsync.js":"/**\r\n * Elle nous permet de se passer du try and catch block ds nos fnction et de la maniere autonome de gerer les erreurs\r\n */\r\n\r\nmodule.exports = fn => {\r\n\r\n    return (req, res, next) => {\r\n        fn(req , res, next).catch(next)\r\n    }\r\n}"},"lineCount":null}},"error":null,"hash":"552c72f14e4e26bb36a41d7e517e578d","cacheData":{"env":{}}}